src/components/include/utils/threads/thread.h

```diff
@@ -114,6 +114,7 @@ class Thread {
   // Should be locked to protect isThreadRunning_ and thread_created_ values
   sync_primitives::Lock state_lock_;
   volatile bool isThreadRunning_;
+  volatile bool do_iteration_;
```

Comment:
@anosach-luxoft, fixed in [ce7130e](https://github.com/smartdevicelink/sdl_core/pull/707/commits/ce7130e0fa97a34d5fb143ff0764557cee82eb68)